risc0/zkvm/sdk/rust/src/build/rcc.rs

```diff
@@ -0,0 +1,614 @@
+#![cfg(feature = "risc_cc")]
+
+use std::{env, ffi::OsStr, path::Path};
+
+use cc;
+use cc::Error;
+
+#[allow(missing_docs)]
+pub const DEFAULT_RV_CC_INSTALL_PATH: &str = "/opt/riscv";
+
+#[derive(Debug)]
+/// A convenient wrapper around `cc::Build` with sensible defaults for building
+/// to `riscv32im` targets. See `./mod.rs` for further explanation.
+pub struct Build {
+    inner: cc::Build,
+    rv_cc_install_path: String,
+    no_risc0_default_flags: bool,
+    compiler_default_flags: bool,
+    release: bool,
+}
+
+impl Build {
+    /// Construct a new instance with the default riscv32 configuration.
+    ///
+    /// This builder is finished with the [`compile`] function.
+    ///
+    /// [`compile`]: struct.Build.html#method.compile
+    pub fn new() -> Self {
+        let is_riscv_env = env::var("CARGO_CFG_TARGET_ARCH").unwrap() == "riscv32";
+
+        Self {
+            inner: cc::Build::new(),
+            rv_cc_install_path: DEFAULT_RV_CC_INSTALL_PATH.to_string(),
+            no_risc0_default_flags: !is_riscv_env,
+            compiler_default_flags: false,
+            release: false,
+        }
+    }
+
+    /// Add a directory to the `-I` or include path for headers
+    pub fn include<P: AsRef<Path>>(&mut self, dir: P) -> &mut Self {
+        self.inner.include(dir);
+        self
+    }
```

Comment:
Instead of passing thru everything (and thus adding lots of boilerplate to this crate, and the need to document it all and maintain it), I think we could just expose the `inner: cc::Build` directly.

Reply:
I'm a bit unclear how we could do that without extending `inner::cc::Build` (as Rust doesn't seem to support extending structs)

Any suggestions?

I guess, if we're okay with giving up the environment variables specific for this object such as `RCC_COMPILER` and `RV_CC_INSTALL_PATH` we could have a constructor that applies the default risc0 flags and returns the inner Build directly